tcs34725.py

```diff
@@ -123,22 +123,17 @@ def _temperature_and_lux(self, data):
         cct = 449.0 * n**3 + 3525.0 * n**2 + 6823.3 * n + 5520.33
         return cct, y
 
-    def html_rgb(self):
-        r, g, b, c = self.read(raw=True)
-        red = int((r/c) * 256)
-        green = int((g/c) * 256)
-        blue = int((b/c) * 256)
-
-        gammatable = {}
-        for x, y in enumerate(range(0, 256)):
-            x = x / 255
-            x = pow(x, 2.5)
-            x = x * 255
-            gammatable[y] = x
-        rgbhex = "{0:02x}{1:02x}{2:02x}".format(int(gammatable[red]),
-                                 int(gammatable[green]),
-                                 int(gammatable[blue]))
-        return rgbhex
+    def html_rgb(self, data):
+
+        r, g, b, c = data
+        red = pow((int((r/c) * 256) / 255), 2.5) * 255
+        green = pow((int((g/c) * 256) / 255), 2.5) * 255
+        blue = pow((int((b/c) * 256) / 255), 2.5) * 255
+
+        rgbhex = "{0:02x}{1:02x}{2:02x}".format(int(red),
+                                 int(green),
+                                 int(blue))
+        return red, green, blue, rgbhex
```

Comment:
It doesn't need to be a method anymore. Can you make it a normal function?

Also, I would make the html format a separate function after all, possibly calling the rgb function internally. The reason is, you really ever need one or the other, never both.